remotes/transport/http.go

```diff
@@ -0,0 +1,223 @@
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		flags := feature.FlagsFromContext(r.Context())
+
+		if flagVal, ok := flags[feature.ReplicationStreamBackend().Key()]; !ok || !flagVal.(bool) {
+			h.api.Respond(w, r, http.StatusNotFound, nil)
+			return
+		}
+
+		next.ServeHTTP(w, r)
+	})
+}
+
+func (h *RemoteConnectionHandler) handleGetRemotes(w http.ResponseWriter, r *http.Request) {
+	q := r.URL.Query()
+
+	// orgID is required for listing remotes.
+	orgID := q.Get("orgID")
+	o, err := platform.IDFromString(orgID)
+	if err != nil {
+		h.api.Err(w, r, errBadOrg)
+		return
+	}
+
+	// name and remoteURL are optional additional filters.
+	name := q.Get("name")
+	remoteURL := q.Get("remoteURL")
+
+	filters := influxdb.RemoteConnectionListFilter{OrgID: *o}
+	if name != "" {
+		filters.Name = &name
+	}
+	if remoteURL != "" {
+		filters.RemoteURL = &remoteURL
+	}
+
+	rs, err := h.remotesService.ListRemoteConnections(r.Context(), filters)
+	if err != nil {
+		h.api.Err(w, r, err)
+		return
+	}
+	h.api.Respond(w, r, http.StatusOK, rs)
+}
+
+func (h *RemoteConnectionHandler) handlePostRemote(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	q := r.URL.Query()
+
+	validate := q.Get("validate") == "true"
+	var req influxdb.CreateRemoteConnectionRequest
+	if err := h.api.DecodeJSON(r.Body, &req); err != nil {
+		h.api.Err(w, r, err)
+		return
+	}
+
+	if validate {
+		if err := h.remotesService.ValidateNewRemoteConnection(ctx, req); err != nil {
+			h.api.Err(w, r, err)
+			return
+		}
+		h.api.Respond(w, r, http.StatusNoContent, nil)
+		return
+	}
+
+	remote, err := h.remotesService.CreateRemoteConnection(ctx, req)
+	if err != nil {
+		h.api.Err(w, r, err)
+		return
+	}
+	h.api.Respond(w, r, http.StatusCreated, remote)
+}
+
+func (h *RemoteConnectionHandler) handleGetRemote(w http.ResponseWriter, r *http.Request) {
+	id, err := platform.IDFromString(chi.URLParam(r, "id"))
+	if err != nil {
+		h.api.Err(w, r, errBadId)
+		return
+	}
+
+	remote, err := h.remotesService.GetRemoteConnection(r.Context(), *id)
+	if err != nil {
+		h.api.Err(w, r, err)
+		return
+	}
+	h.api.Respond(w, r, http.StatusOK, remote)
+}
+
+func (h *RemoteConnectionHandler) handlePatchRemote(w http.ResponseWriter, r *http.Request) {
+	id, err := platform.IDFromString(chi.URLParam(r, "id"))
+	if err != nil {
+		h.api.Err(w, r, errBadId)
+		return
+	}
+
+	ctx := r.Context()
+	q := r.URL.Query()
+
+	validate := q.Get("validate") == "true"
+	var req influxdb.UpdateRemoteConnectionRequest
+	if err := h.api.DecodeJSON(r.Body, &req); err != nil {
+		h.api.Err(w, r, err)
+		return
+	}
+
+	if validate {
+		if err := h.remotesService.ValidateUpdatedRemoteConnection(ctx, *id, req); err != nil {
+			h.api.Err(w, r, err)
+			return
+		}
+		h.api.Respond(w, r, http.StatusNoContent, false)
+		return
+	}
+
+	remote, err := h.remotesService.UpdateRemoteConnection(r.Context(), *id, req)
```

Comment:
`ctx` is already available as a variable from line 170 so I think you could use it here instead of an additional `r.Context()` call?